controllers/update_pods.go

```diff
@@ -249,22 +249,10 @@ func deletePodsForUpdates(ctx context.Context, r *FoundationDBClusterReconciler,
 		}
 	}
 
+	// TODO(jscheuermann): If a maintenance zone is already set, we should allow to delete Pods in this zone.
 	if deletionMode == fdbv1beta2.PodUpdateModeZone && cluster.UseMaintenaceMode() {
-		var processGroups []string
-		for _, pod := range deletions {
-			processGroups = append(processGroups, pod.Labels[cluster.GetProcessGroupIDLabel()])
-		}
-
 		logger.Info("Setting maintenance mode", "zone", zone)
-		cluster.Status.MaintenanceModeInfo = fdbv1beta2.MaintenanceModeInfo{
-			StartTimestamp: &metav1.Time{Time: time.Now()},
-			ZoneID:         zone,
-			ProcessGroups:  processGroups,
-		}
-		err = r.updateOrApply(ctx, cluster)
-		if err != nil {
-			return &requeue{curError: err}
-		}
+
```

Comment:
We should add a check here to only set the maintenance mode if at least 1 storage process is included in the list. Otherwise the maintenance mode will be useless, as it only works on storage processes.